src/downshift.js

```diff
@@ -587,44 +604,15 @@ class Downshift extends Component {
 
   /////////////////////////////// LABEL
 
-  getLabelProps = (props = {}) => {
-    this.getLabelProps.called = true
-    if (
-      this.getInputProps.called &&
-      props.htmlFor &&
-      props.htmlFor !== this.inputId
-    ) {
-      throw new Error(
-        `downshift: You provided the htmlFor of "${
-          props.htmlFor
-        }" for your label, but the id of your input is "${
-          this.inputId
-        }". You must either remove the id from your input or set the htmlFor of the label equal to the input id.`,
-      )
-    }
-    this.inputId = firstDefined(this.inputId, props.htmlFor, `${this.id}-input`)
-    return {
-      ...props,
-      htmlFor: this.inputId,
-    }
```

Comment:
Seriously simplified things by adding the new props. We no longer validate for you, but you can set your ids earlier on and we don't have to validate. You can still set ids directly on elements like before, but you wont get validation anymore. I think that's an ok tradeoff.